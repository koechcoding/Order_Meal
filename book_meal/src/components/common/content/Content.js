import React from 'react';
import PropTypes from 'prop-types';
import menuImg from './assets/menu.png';

const Content = ({ children, contentTop, contentFilter, toggleMnu })=>(
    <div className="col-12 col-md-8 offset-md-4 col-lg-9 offset-lg-3 container-fluid p-4">
        <div className='row'>
            <div className='col-12'>
                <img className='d-md-none controls-icon pb-3'
                   onClick={toggleMenu}
                   alt="Menu"
                   src={menuImg}/>
            </div>
            {content Top}
        </div>
        {contentFilter}
        <div className="row">
            <div className='col-12'>
                {children}
            </div>
        </div>
    </div>
);

Content.propTypes = {
    children: PropTypes.array,
    contentTop: PropTypes.element,
    contentFilter: PropTypes.element,
    toggleMenu: PropTypes.func.isRequired,
};

export default Content;