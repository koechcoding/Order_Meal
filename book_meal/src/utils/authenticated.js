export default function authenticated(){
    try{
        return JSON.parse(localStorage.getItem('user'));
    } catch(err){
        
    }
}