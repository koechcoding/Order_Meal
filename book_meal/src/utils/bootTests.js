import Enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";
Enzyme.configure({ adapter: new Adapter()});

global.window = {}
import localStorage from 'mock-local-Storage';
window.localStorage = global.localStorage

export default Enzyme;
