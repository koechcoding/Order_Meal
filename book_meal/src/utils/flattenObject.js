export default function flattenObject(obj){
    let result = {}
    for (const key in obj){
        if((typeof obj[key])=== 'object' &&  obj[key]  !== null) {
            let flattened = flattenObject(obj[key])
            for(const key2 in flattened){
                result[key + "." + key2] = flattened[key2];
            }
        }else{
            result[key] = obj[key];
        }
    }
    return result;
}