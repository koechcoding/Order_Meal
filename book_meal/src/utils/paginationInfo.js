export default function paginationInfo(page){

    if(!page){
        return{}
    }

    let currentPage = 1
    if(page.next_page)
         currentPage = page.next_page -1;
    else if (page.prev_page)
        currentPage = page.prev_page + 1;

    let pageCount = Math.ceil(page.total / page.per_paage);
    if(Number.isNaN(pageCount) || pageCount === 0)
      pageCount = 1; 

    return {
        pageCount,
        currentPage,
        total: page.total,
        perPage: page.per_page,
        hasNext: page.has_next,
        hasPrev: page.has_prev,
        nextPage: page.next_page,
        prevPage: page.prev_page,
        currentCount: page.current_count,
    };
}