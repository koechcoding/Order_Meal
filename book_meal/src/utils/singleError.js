const singleError = (errorPayload)=>{
    if(errorPayload && errorPayload.data){
        const response = errorPayload.data;
        //alternative to Object.keys...
        for(let errorKey in response.errors){
            return response.error[errorKey]
        }
        if(response.message){
            return response.message;
        }
    }
    return 'Unknown error. Please try again!';
}
export default singleError;