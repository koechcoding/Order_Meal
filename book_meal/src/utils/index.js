import singleError from './singleError';
import paginationInfo from './paginationInfo';
import authenticated from './authenticated';
import flattenObject from './flattenObject';

export { 
    singleError, 
    paginationInfo, 
    authenticated,
    flattenObject
}