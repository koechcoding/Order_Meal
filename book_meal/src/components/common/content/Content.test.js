import React from 'react';
import { shallow } from 'enzyme';
import 'src/utils/bootTests'; //order matters
import Content from './Content';
import Adapter from 'enzyme-adapter-react-16';


describe('<Content/>', ()=>{
    it('renders correctly', ()=>{
        const wrapper = shallow(<content toggleMenu={()=>{}}/>);
        expect(wrapper).toMatchSnapshot();
    });
});