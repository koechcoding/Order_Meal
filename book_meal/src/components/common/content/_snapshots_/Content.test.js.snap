//Jest snapshot v1, https://goo.gl/fbAQLP


exports[`<Content /> renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Content
    toggleMenu={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="row"
>
          <div
                    className="col-12"
          >
                    <img
                              alt="Menu"
                              className="d-md-none controls-icon pb-3"
                              onClick={[Function]}
                              src="menu.png"
                    />
          </div>
</div>,
        undefined,
        <div
          className="row"
>
          <div
                    className="col-12"
          />
</div>,
      ],
      "className": "col-12 col-md-8 offset-md-4 col-lg-9 offset-lg-3 container-fluid p-4",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="col-12"
>
              <img
                            alt="Menu"
                            className="d-md-none controls-icon pb-3"
                            onClick={[Function]}
                            src="menu.png"
              />
</div>,
            undefined,
          ],
          "className": "row",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <img
                alt="Menu"
                className="d-md-none controls-icon pb-3"
                onClick={[Function]}
                src="menu.png"
/>,
              "className": "col-12",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "Menu",
                "className": "d-md-none controls-icon pb-3",
                "onClick": [Function],
                "src": "menu.png",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            "type": "div",
          },
          undefined,
        ],
        "type": "div",
      },
      undefined,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="col-12"
/>,
          "className": "row",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "col-12",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="row"
>
            <div
                        className="col-12"
            >
                        <img
                                    alt="Menu"
                                    className="d-md-none controls-icon pb-3"
                                    onClick={[Function]}
                                    src="menu.png"
                        />
            </div>
</div>,
          undefined,
          <div
            className="row"
>
            <div
                        className="col-12"
            />
</div>,
        ],
        "className": "col-12 col-md-8 offset-md-4 col-lg-9 offset-lg-3 container-fluid p-4",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="col-12"
>
                <img
                                alt="Menu"
                                className="d-md-none controls-icon pb-3"
                                onClick={[Function]}
                                src="menu.png"
                />
</div>,
              undefined,
            ],
            "className": "row",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <img
                  alt="Menu"
                  className="d-md-none controls-icon pb-3"
                  onClick={[Function]}
                  src="menu.png"
/>,
                "className": "col-12",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "Menu",
                  "className": "d-md-none controls-icon pb-3",
                  "onClick": [Function],
                  "src": "menu.png",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              "type": "div",
            },
            undefined,
          ],
          "type": "div",
        },
        undefined,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="col-12"
/>,
            "className": "row",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
              "className": "col-12",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;